### PR TITLE
Add definitions and profiles to ZYYX+, ZYYX Pro and ZYYX Pro ii

### PR DESCRIPTION
**Pull request by theodorhansson**
# Description

This adds three printers that were previously missing from Cura. 

* ZYYX+
* ZYYX Pro
* ZYYX Pro ii

This also changes the manufacturer of the following printer. Which was previously made by Magicfirm Europe, but the firm was rebranded to ZYYX Labs AB.
* ZYYX Agile

## Type of change

Implements 

- [x] Printer definition file(s)

# How Has This Been Tested?

Testprints have been performed on all ZYYX printers except ZYYX Agile.

**Test Configuration**:
- [x] Windows
- [ ] Linux (the profiles work but compiling to .x3g does not work on all machines)
- [x] MacOS 

# Checklist:

- [x] My code follows the style guidelines of this project as described in [UltiMaker Meta](https://github.com/Ultimaker/Meta) and [Cura QML best practices](https://github.com/Ultimaker/Cura/wiki/QML-Best-Practices)
- [x] I have read the [Contribution guide](https://github.com/Ultimaker/Cura/blob/main/CONTRIBUTING.md) 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have uploaded any files required to test this change

### CODE REVIEW

**Comment by theodorhansson** [comment]
Note: this PR references some materials not yet implemented in Cura. These materials are currently awating approval in [PR#281 in Ultimaker/fdm_materials](https://github.com/Ultimaker/fdm_materials/pull/281).

**Comment by DavidPrytzArcombe** [comment]
I tested one of these profilers on a YYX+, it worked well and sped up the print. Would recommend.

**Comment by theodorhansson** [comment]
The profiles in this PR have been beta-tested by some different users of ZYYX printers and I feel that the profiles are now ready for review. The profiles were distributed to the testers from ([theodorhansson/ZYYX-Cura-Printer-Profiles](https://github.com/theodorhansson/ZYYX-Cura-Printer-Profiles/releases).

It would be appreciated if someone could review the changes proposed.

**Comment by theodorhansson** [comment]
@saumyaj3 @nallath @HellAholic @wawanbreton

**Comment by wawanbreton** [comment]
Hi @theodorhansson, this is quite a big PR, but it looks good at first glance. I will take some time to review it. What about the "Agile" printer then ? If this is no official product, I would indeed recommend to remove it, otherwise your users may get confused or think that this is an incoming printer.

**Comment by theodorhansson** [comment]
@wawanbreton I do not work for ZYYX, but I have spoken with their CEO over the phone. He claimed that the "Agile" printer was a concept that would evolve into the "Pro". My recommendation is that the "Agile" printer can be removed from Cura.

**Comment by theodorhansson** [comment]
@wawanbreton It's been removed, but the bot complains...

**Comment by wawanbreton** [comment]
Indeed :sweat_smile: I'll see with the team how to handle this properly

**Comment by wawanbreton** [comment]
So, the consequences of removing the printer definition are the following:
* Users having an instance of this printer won't be able to use it anymore
* Project files using this printer won't be usable anymore

The very proper way to handle this situation would have been to create an upgrade script, and convert the printers and project files using Agile to now using Pro, which is what actually happened. However, this is much more complex, so if we assume nobody in the world (except maybe a few engineers at ZYYX) have used this printer, we can safely ignore this warning. Do you think it would be worth contacting them again to check, or can we safely remove it, knowing the consequences ?

**Comment by theodorhansson** [comment]
My view is that very few people use the printer and it would therefore not be worth writing an upgrade script. The ZYYX Pro was released in 2017, so I doubt that even the engineers at ZYYX are still using the "Agile". Furthermore, up until this PR I believe that most users used Simplyfy3D instead of Cura.

**Comment by wawanbreton** [comment]
Ok, looks like they just made a test with Cura, that ended up in the mainstream for weak reasons. I'm good with removing it then. If people somehow get to complain about this, they can still restore the files on their installation, and we shall put it back, but that is extremely unlikely.

File: resources/quality/zyyx_plus/ABS/zyyx_plus_abs_fast.inst.cfg
Comment (wawanbreton) [inline resources/quality/zyyx_plus/ABS/zyyx_plus_abs_fast.inst.cfg:9]: `setting_version` should be 23. If you are actually using old interpretation/naming of settings, they should be manually upgraded. Same for all other files.

File: resources/variants/zyyx/zyyx_pro_carbon0.6.inst.cfg
Comment (wawanbreton) [inline resources/variants/zyyx/zyyx_pro_carbon0.6.inst.cfg:14]: This is not necessary. `line_width` is by default equal to `machine_nozzle_size`, which you already properly define. Same for other variants.

**Review by wawanbreton — CHANGES_REQUESTED** [review]
I spotted one issue with setting_version, and a possible optimization for variants. Otherwise looks very good !

**Comment by theodorhansson** [comment]
@wawanbreton Versions updated and line_width removed!

**Comment by wawanbreton** [comment]
Nice, thanks ! I just saw that some variants defined a line_width different from the machine_nozzle_size tough, those should probably be restored ?

Also there are some conflicts with main, because we just upgraded the settings_version to 24 (didn't think we would merge it that fast :grinning:) so you can ignore the conflicts and force your files removals.

**Comment by wawanbreton** [comment]
> Should I bump the settings version to 24?

As you are on it, you can. Otherwise we plan to make an other pass for it.

**Comment by wawanbreton** [comment]
changes between 23 and 24 are really self-contained and don't concern your settings at all, so you can safely put either 23 or 24

**Comment by theodorhansson** [comment]
Then 23 is ok by me.

**Comment by wawanbreton** [comment]
Hi @theodorhansson, can you just fix the git conflicts ? After that I guess we can merge this PR

**Comment by wawanbreton** [comment]
Very nice, thanks ! I can now approve and it will be merged after further testing.

**Review by wawanbreton — APPROVED** [review]

**Comment by wawanbreton** [comment]
Hi @theodorhansson, we just realized that the profiles contain a flow ratio of 420, which is totally outside usual specs. Is this on purpose, or possibly a mistake ?

**Comment by theodorhansson** [comment]
Hi @wawanbreton! It is on purpose. The extruder was changed by the manufacturer, but the firmware was not. I believe it was a geared Bondtech, but I'm not 100%. Unfortunately I have not personally printed with a ZYYX Pro II.

**Comment by theodorhansson** [comment]
I looked again at the Simplify3D profiles from the manufacturer and they also contain an abnormally large extrusion multiplier.

**Comment by wawanbreton** [comment]
Ok, thanks for your quick answer :slightly_smiling_face:
We have decided to exclude this printer for the moment: the very high flow value may have side-effects on the slicing engine that nobody will see until they start a print. Since we are unable to confirm that this is not the case, we prefer not to risk it for now.
Maybe you can contact ZYYX and ask them for some testing, if they have an interest to it ?

**Pull request merged**
